test/Microsoft.SqlTools.ServiceLayer.Test/QueryExecution/Common.cs

```diff
@@ -96,6 +97,11 @@ public static Query GetBasicExecutedQuery()
         public static Query GetBasicExecutedQuery(QueryExecutionSettings querySettings)
         {
             ConnectionInfo ci = CreateTestConnectionInfo(new[] {StandardTestData}, false);
+
+            // Query won't be able to request a new query DbConnection unless the ConnectionService has a 
+            // ConnectionInfo with the same URI as the query, so we will manually set it
+            ConnectionService.Instance.OwnerToConnectionMap[ci.OwnerUri] = ci;
```

Comment:
I agree. I tried to mess with the services a bit to try and prevent us from needing to do this, but I realized it would be a much larger and annoying change than I hoped.